spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/oauth2/resource/servlet/OAuth2ResourceServerJwtConfiguration.java

```diff
@@ -97,14 +102,14 @@ private void jwsAlgorithms(Set<SignatureAlgorithm> signatureAlgorithms) {
 		}
 
 		private OAuth2TokenValidator<Jwt> getValidators(OAuth2TokenValidator<Jwt> defaultValidator) {
-			List<String> audiences = this.properties.getAudiences();
-			if (CollectionUtils.isEmpty(audiences)) {
-				return defaultValidator;
-			}
 			List<OAuth2TokenValidator<Jwt>> validators = new ArrayList<>();
 			validators.add(defaultValidator);
-			validators.add(new JwtClaimValidator<List<String>>(JwtClaimNames.AUD,
-					(aud) -> aud != null && !Collections.disjoint(aud, audiences)));
+			validators.addAll(this.customOAuth2TokenValidators);
+			List<String> audiences = this.properties.getAudiences();
+			if (!CollectionUtils.isEmpty(audiences)) {
+				validators.add(new JwtClaimValidator<List<String>>(JwtClaimNames.AUD,
+						(aud) -> aud != null && !Collections.disjoint(aud, audiences)));
+			}
 			return new DelegatingOAuth2TokenValidator<>(validators);
```

Comment:
If there are no audiences and no custom validators, we should return the `defaultValidator` rather than creating a `DelegatingOAuth2TokenValidator` with a single delegate.

Reply:
I added a check for number of validators in the list to avoid duplication of the `CollectionUtils.isEmpty(audiences)` check. I think this way it will be easier to add new predefined validators like the one for audiences if needed.